src/Radio/index.js

```diff
@@ -7,21 +7,26 @@ import { defaultTheme } from '../Theme'
 const Root = styled.span`
   position: relative;
   display: block;
-
-  opacity: ${props => (props.readonly || props.disabled ? 0.35 : 1)};
-  cursor: ${props => (props.readonly || props.disabled ? 'not-allowed' : 'default')};
+  opacity: ${props => (props.disabled ? 0.6 : 1)};
 `
 
 const Label = styled.label`
   display: inline-flex;
   align-items: center;
   line-height: 1;
+  cursor: ${props => (props.readonly || props.disabled ? 'not-allowed' : 'default')};
+  border: ${props =>
+    props.viewOnly && (props.checked || props.defaultChecked) ? '2px solid #999' : '0'};
+  border-width: ${props =>
+    props.viewOnly && (props.checked || props.defaultChecked) ? '0 2px' : '0'};
+  border-radius: 10px;
 `
 
 const Svg = styled.svg`
+  display: ${props => (props.readonly || props.viewOnly ? 'none' : 'default')};
   width: ${props => (props.forPrint ? '1em' : '28px')};
   height: ${props => (props.forPrint ? '1em' : '28px')};
-  margin-top: 8px;
+  margin-top: 8px; // TODO: remove this?
```

Comment:
Yeah, I think we probably should, that's just my opinion though

Reply:
I pulled it out in the v3 branch.  Some stuff definitely made it into some of these components that was inherited from a specific app implementation.